tools/scripts/Makefile

```diff
@@ -0,0 +1,265 @@
+	  -mfpu=vfpv3 							\
+	  -mfloat-abi=hard
+
+LDFLAGS += -specs=$(BUILD_DIR)/app/src/Xilinx.spec 			\
+	   -mfpu=vfpv3							\
+ 	   -mfloat-abi=hard 						\
+	   -mcpu=cortex-a9						\
+	   -Wl,-build-id=none
+endif
+# Zynqmp 
+ifeq (psu_cortexa53_0,$(strip $(PROCESSOR_CORE)))
+
+CC = aarch64-none-elf-gcc
+
+endif
+# Microblaze
+ifeq (sys_mb,$(strip $(PROCESSOR_CORE)))
+
+CC = microblaze-xilinx-elf-gcc
+
+CFLAGS += -mcpu=cortex-a9 						\
+	  -DXILINX -DMICROBLAZE						\
+	  -DXILINX_PLATFORM						\
+	  -mlittle-endian						\
+	  -mxl-barrel-shift						\
+	  -mxl-pattern-compare 						\
+	  -mno-xl-soft-div						\
+	  -mcpu=v11.0							\
+	  -mno-xl-soft-mul						\
+	  -mxl-multiply-high
+
+LDFLAGS += -Xlinker --defsym=_HEAP_SIZE=0x100000 			\
+	   -Xlinker --defsym=_STACK_SIZE=0x2000 			\
+	   -mlittle-endian 						\
+	   -mxl-barrel-shift						\
+	   -mxl-pattern-compare						\
+	   -mno-xl-soft-div						\
+	   -mcpu=v11.0							\
+	   -mno-xl-soft-mul						\
+	   -mxl-multiply-high 						\
+	   -Wl,--no-relax 						\
+	   -Wl,--gc-sections 
+
+endif
+
+# Common xilinx libs
+LIBS =	-Wl,--start-group,-lxil,-lgcc,-lc,--end-group
+
+endif
+
+# Add the common include paths
+INC_PATHS = $(foreach x, $(INCS), $(addprefix -I ,$(dir $(x))))
+LIB_PATHS = $(foreach x, $(LIBRARY_DIR), $(addprefix -L ,$(x)))
+EXTRA_INCS =
+
+OBJS = $(SRCS:.c=.o)
+
+# Compile all the source files
+%.o : %.c
+	$(ECHO) [CC]	$(notdir $<) '\n'
+	$(MUTE)$(CC) $(CFLAGS) $(INC_PATHS) $(EXTRA_INCS)		\
+	-c $(addprefix $(BUILD_DIR)/app/src/,$(notdir $<))		\
+	-o $(addprefix $(OBJECTS_DIR)/,$(notdir $@))
+
+# Link the resulted object files
+$(EXEC): check-srcs $(subst LOCAL_PLATFORM,xilinx,$(OBJS))		\
+	$(BUILD_DIR)/app/src/lscript.ld
+	$(ECHO) [LD] $(shell ls $(OBJECTS_DIR)) '\n'
+	$(MUTE)$(CC) $(LDFLAGS) $(LIB_PATHS) 				\
+	$(OBJECTS_DIR)/*.o						\
+	$(LIBS) -o $(BUILD_DIR)/$(EXEC).elf
+
+# Copy the source files from the no-Os repo to the local project
+define copy_srcs
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(SRCS))		\
+	$(BUILD_DIR)/app/src/
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(INCS))		\
+	$(BUILD_DIR)/app/src/
+endef
+
+.DEFAULT_GOAL := all
+all: 	xilinx
+
+compile: $(EXEC)
+
+create-obj-dir:
+	$(MUTE)mkdir -p $(OBJECTS_DIR)
+	
+create-build-dir:
+	$(MUTE)mkdir -p $(BUILD_DIR)
+
+copy-srcs:
+	$(call copy_srcs,$(PLATFORM))
+
+check-srcs:
+ifeq (xilinx,$(strip $(PLATFORM)))
+# Replace the target sources (the default ones are the repository files) with
+# the local project sources
+	$(eval SRCS :=$(shell find $(BUILD_DIR)/app/src/ -name '*.c'))
+	$(eval EXTRA_INCS :=-I $(BUILD_DIR)/bsp/$(PROCESSOR_CORE)/include)
+	$(eval INC_PATHS :=-I $(BUILD_DIR)/app/src)
+endif
+
+clean:
+	$(ECHO) Cleaning build workspace '\n'
+	$(MUTE)rm -rf temp
+	$(MUTE)rm -rf $(BUILD_DIR)
+	@$(MAKE) -s xil-pre-cleanup
+
+eval-hardware:
+# If the hardware file is not specified, start searching for one
+# Check for .hdf files inside the project directory		
+ifndef $(HARDWARE)
```

Comment:
why not move everything into a shell script and do `PLATFORM=$(shell script,param1,param2,etc)` ?

Reply:
Since the makefile offers this functionality, why not use it. Also, in this way, the code is a little bit more organized and not split in several files that will be shared only to this makefile.